ln-gateway/src/webserver.rs

```diff
@@ -29,17 +29,18 @@ pub async fn pay_invoice(
         .await
         .expect("failed to send over channel");
     receiver.await.unwrap()?;
-
-    Ok(Response::new(200))
+    Ok(())
 }
 
-pub async fn run_webserver(sender: mpsc::Sender<GatewayRequest>) -> tide::Result<()> {
-    // Tide state must be Sync
```

Comment:
It looks like Axum doesn't require the state to be Sync. I was able to just use Extension<mpsc::Sender<GatewayRequest>> instead of Extension<Arc<Mutex<mpsc::Sender<GatewayRequest>>>>. This is nice because it gets rid of Arcs and Mutexs.

Reply:
The reason for not requiring the mutex anymore is actually a different one.

While [`Extension<T>`](https://docs.rs/axum/latest/axum/struct.Extension.html#) itself does not place any trait bounds on `T`, the [implementation of `Layer`](https://docs.rs/axum/latest/axum/struct.Extension.html#impl-Layer%3CS%3E-for-Extension%3CT%3E) does. This also makes sense given that the state has to be shared between threads when processing requests in parallel.

The reason why we do not need the mutex anymore is that [`Sender`](https://docs.rs/tokio/latest/tokio/sync/mpsc/struct.Sender.html) itself is a synchronization primitive and [implements `Send` and `Sync`](https://docs.rs/tokio/latest/tokio/sync/mpsc/struct.Sender.html#impl-Sync-for-Sender%3CT%3E).

Reply:
Removed it. I guess we don't need Arc because `Sender` implements `Clone` ?